fpdf/text_region.py

```diff
@@ -131,6 +132,96 @@ def build_lines(self, print_sh) -> List[LineWrapper]:
         return text_lines
 
 
+class ImageParagraph:
+    def __init__(
+        self,
+        region,
+        name,
+        align=None,
+        width: float = None,
+        height: float = None,
+        fill_width: bool = False,
+        keep_aspect_ratio=False,
+        top_margin=0,
+        bottom_margin=0,
+        link=None,
+        title=None,
+        alt_text=None,
+    ):
+        self.region = region
+        self.name = name
+        if align:
+            align = Align.coerce(align)
+        self.align = align
+        self.width = width
+        self.height = height
+        self.fill_width = fill_width
+        self.keep_aspect_ratio = keep_aspect_ratio
+        self.top_margin = top_margin
+        self.bottom_margin = bottom_margin
+        self.link = link
+        self.title = title
+        self.alt_text = alt_text
+
+    def build_line(self):
+        # We do double duty as a "text line wrapper" here, since all the necessary
+        # information is already in the ImageParagraph object.
+        self.name, self.img, self.info = self.region.pdf.preload_image(self.name, None)
+        return self
+
+    def render(self, col_left, col_width, max_height):
+        is_svg = isinstance(self.info, VectorImageInfo)
+        if self.height:
+            h = self.height
+        else:
+            native_h = self.info["h"] / self.region.pdf.k
+        if self.width:
+            w = self.width
+        else:
+            native_w = self.info["w"] / self.region.pdf.k
+            if native_w > col_width or self.fill_width:
+                w = col_width
+            else:
+                w = native_w
+        if not self.height:
+            h = w * native_h / native_w
+        if h > max_height:
+            return None
+        x = col_left
+        if self.align:
+            if self.align == Align.R:
+                x += col_width - w
+            elif self.align == Align.C:
+                x += (col_width - w) / 2
```

Comment:
In `table.py` we raise a `ValueError` if a value of `Align.J` is provided.
We should probably do the same there.

Reply:
Good idea. We could probably extend that to `Align.X` as well (= anything other than L/C/R).